opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/CohortManager.java

```diff
@@ -769,6 +791,32 @@ private MyResourceId commonGetAnnotationSet(String id, @Nullable String studyStr
         return cohortAclList;
     }
 
+    public List<QueryResult<CohortAclEntry>> getAcl(String studyStr, List<String> cohortList, String member,
+                                                    boolean silent, String sessionId)
+            throws CatalogException {
+        ParamUtils.checkObj(member, "member");
+
+        MyResourceIds resource = getIds(cohortList, studyStr, sessionId);
+        checkMembers(resource.getStudyId(), Arrays.asList(member));
+
+        List<QueryResult<CohortAclEntry>> cohortAclList = new ArrayList<>(resource.getResourceIds().size());
+        for (Long cohortId : resource.getResourceIds()) {
+            try {
+                QueryResult<CohortAclEntry> allCohortAcls =
+                        authorizationManager.getCohortAcl(resource.getStudyId(), cohortId, resource.getUser(), member);
+                allCohortAcls.setId(String.valueOf(cohortId));
+                cohortAclList.add(allCohortAcls);
+            } catch (CatalogException e) {
+                if (silent) {
+                    cohortAclList.add(new QueryResult<>(studyStr, 0, 0, 0, "", e.toString(), new ArrayList<>(0)));
```

Comment:
Here, we should be returning the id like it was explained before. In any case, because these two methods are that similar, I would try to merge them using always this signature. Only if member is not empty we will call to getAllCohortAcls, otherwise to getCohortAcl.